### PR TITLE
enumerating deprecated infrastructure IP space

### PR DESCRIPTION
**Pull request by mabrown-corp**
This is an enumeration of all of the deprecated Fastly anycast or
site-specific IPs which may have ever been exposed (via DNS maps) or used for
GitHub user pages.

This commit also removes a single IP (45.32.88.68) which is not a Fastly IP
and appears to be assigned to Choopa / Vultr Holdings (as of 2017-03-03, this
is originated in 45.32.64.0/19 by AS 20473.)

### CODE REVIEW

**Comment by mabrown-corp** [comment]
Greetings Ben et alia,

Please let me know if there are other documentation details you would like me to submit along with the included set of IPs in this pull request.

-Martin

**Comment by aspires** [comment]
cc @benbalter @parkr

**Comment by benbalter** [comment]
@mabrown-corp This is awesome (and extremely helpful).

One question: Would it be more efficient to store these IPs in CIDR notation?

**Comment by mabrown-corp** [comment]
Perhaps it would be more efficient if you have hundreds or thousands more to add to the LEGACY_IP_ADDRESSES data structure, but this is not a terribly long list, so I'm not unduly worried about it myself.  (I confess I don't know Ruby, nor have I studied exactly how you are using the IPs in this list, so I may not be the best judge here.)

**Comment by mabrown-corp** [comment]
So, an alternate option for restructuring is to remove everything except 151.101.0.0/16 (in use) and 199.232.0.0/16 (planned) from the file `config/fastly-ips.txt`.  Those two /16s are prefixes devoted to providing services to HTTP/HTTPS requesters.  

We use the other prefixes for a variety of different purposes (origin pull, DNS, peering ....) so they are included in our public-ip-list ( https://api.fastly.com/public-ip-list ).

**Comment by benbalter** [comment]
> but this is not a terribly long list, so I'm not unduly worried about it myself

Makes sense. Assumed you'd know the IP space better so and figured I'd ask. Will get this merged and out into production. Thanks for the PR. 🎉

File: lib/github-pages-health-check/domain.rb [hunk 1]
```diff
@@ -14,18 +14,54 @@ class Domain < Checkable
...
         "23.235.39.133",
+        "23.235.40.133",
+        "23.235.41.133",
         "23.235.43.133",
         "23.235.44.133",
+        "23.235.45.133",
         "23.235.46.133",
         "23.235.47.133",
-        "45.32.88.68",
+        "23.235.47.133",
```
Comment (miyagawa) [inline lib/github-pages-health-check/domain.rb]: seems to be duplicate.

**Pull request merged**